frontend/public/components/utils/use-url-hook.ts

```diff
@@ -0,0 +1,46 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import { useCallback, useState } from 'react';
+import { usePoll, useSafeFetch } from '.';
+
+export const URL_POLL_DEFAULT_DELAY = 15000; // 15 seconds
+
+export const useURLPoll = (
```

Comment:
we could improve the types with some generics. Like defining the return value to not be `any`

```suggestion
export const useURLPoll = (export const useURLPoll = <R>(
  url: string,
  delay = URL_POLL_DEFAULT_DELAY,
  ...dependencies: any[]
): URLPoll<R> => {
```

Reply:
Nice !